src/main/java/org/assertj/core/api/AbstractMapAssert.java

```diff
@@ -1153,4 +1157,39 @@ public SELF withThreadDumpOnError() {
   public AbstractObjectArrayAssert<?, Object> extracting(String... keys) {
     return super.extracting(keys);
   }
+
+  /**
+   * Flattens a list of values for given keys from the map under test into an array, this new array becoming
+   * the object under test.
+   * <p>
+   * For example, if you specify "id", "name" and "email" keys then the array will contain the map values for 
+   * these keys, you can then perform array assertions on the extracted values.
+   * <p>
+   * If a given key is not present in the map under test, a null value is extracted.
+   * <p>
+   * Example:
+   * <pre><code class='java'> Map&lt;String, List&lt;String&gt; map = new HashMap&lt;&gt;(); 
+   * List<String> nameList = newArrayList("Dave", "Jeff");
+   * List<String> jobList = newArrayList("Plumber", "Builder");
+   * List<String> cityList = newArrayList("Dover", "Boston");
+   * testMap.put("name", nameList);
+   * testMap.put("job", jobList);
+   * testMap.put("city", cityList); 
+   * assertThat(map)
+      .flatExtracting("name","job","city")
+      .contains("Dave","Jeff","Plumber","Builder","Dover","Boston"); </code></pre>     
+   * <p>
+   * Note that the order of extracted keys value is consistent with the iteration order of the array under test.
+   * <p>
+   * Nested keys are not yet supported, passing "name.first" won't get a value for "name" and then try to extract 
+   * "first" from the previously extracted value, instead it will simply look for a value under "name.first" key.  
```

Comment:
Unless I'm missing something this comment is not relevant.

Reply:
Actually this comment does not make sense for maps (for Object it does though), let's remove it.